hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/PromoteLiteMemberTest.java

```diff
@@ -374,6 +376,110 @@ private void memberAttributes_arePreserved_afterPromotion(boolean isMaster) thro
         }
     }
 
+    @Test
+    public void test_lite_member_promotion_causes_no_data_loss_on_three_members() throws InterruptedException {
+        final int entryCount = 1000;
+
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory();
+        Config config = new Config().setLiteMember(true);
+
+        // start first hazelcast instance as a lite member
+        final HazelcastInstance firstHazelcastInstance = factory.newHazelcastInstance(config);
+
+        // start second and third hazelcast instances as a lite member
+        final HazelcastInstance secondHazelcastInstance = factory.newHazelcastInstance(config);
+        final HazelcastInstance thirdHazelcastInstance = factory.newHazelcastInstance(config);
+
+        // promote all instances to data members
+        firstHazelcastInstance.getCluster().promoteLocalLiteMember();
+        secondHazelcastInstance.getCluster().promoteLocalLiteMember();
+        thirdHazelcastInstance.getCluster().promoteLocalLiteMember();
+
+        // check if cluster is in a good shape
+        assertTrueEventually(new AssertTask() {
```

Comment:
nit: could use `waitClusterForSafeState` here and `assertEqualsEventually` in subsequent `assertTrueEventually` usages below.